File: lab1_2_12/5.py
import sys

def dfs(adj, src, vis, res):
    vis[src] = 1
    res.append(src)
    if src in adj:
        for ngb in adj[src]:
            if not vis[ngb]:
                dfs(adj, ngb, vis, res)

def main():
    try:
        input_data = sys.stdin.read().split()
        if not input_data: return
        it = iter(input_data)
        
        n = int(next(it))
        m = int(next(it))
        
        adj = {}
        for _ in range(m):
            x = int(next(it))
            y = int(next(it))
            if x not in adj: adj[x] = []
            if y not in adj: adj[y] = []
            adj[x].append(y)
            adj[y].append(x)

        vis = [0] * (n + 1)
        res = []
        
        # Original code starts DFS from node 1
        dfs(adj, 1, vis, res)

        print(*(res))
    except (StopIteration, ValueError, EOFError):
        pass

if __name__ == "__main__":
    main()

File: lab1_2_12/5.cpp
#include<bits/stdc++.h>
using namespace std;

void dfs(vector<int> adj[], int src, vector<int>& vis, vector<int>& res)
{
    vis[src] = 1;
    res.push_back(src);
    for(auto ngb : adj[src])
    {
        if(!vis[ngb])
            dfs(adj, ngb, vis, res);
    }
}


int main()
{
    cout << "Enter no of vertices & edges : ";
    int n,m;
    cin >> n >> m;
    cout << "Enter each edge's src and dst node\n";
    vector<int> adj[n+1];
    for(int i=0;i<m;i++)
    {
        int x,y;
        cin >> x >> y;
        adj[x].push_back(y);
        adj[y].push_back(x);
    }

    vector<int> vis(n+1, 0);
    vector<int> res;
    dfs(adj, 1, vis, res);

    for(int i= 0;i<res.size();i++)
        cout << res[i] << " ";
    
    cout << "\n";

    return 0;
}